libparsec/crates/platform_device_loader/src/lib.rs

```diff
@@ -396,6 +394,64 @@ fn load_available_device_from_blob(
     })
 }
 
+#[cfg_attr(target_arch = "wasm32", expect(dead_code))]
+fn encrypt_device(device: &LocalDevice, key: &SecretKey) -> Bytes {
+    let mut cleartext = zeroize::Zeroizing::new(device.dump());
+    let ciphertext = key.encrypt(&cleartext);
+    cleartext.zeroize();
+    ciphertext.into()
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum DecryptDeviceFileError {
+    #[error("Failed to decrypt device file: {0}")]
+    Decrypt(CryptoError),
+    #[error("Failed to load device: {0}")]
+    Load(&'static str),
+}
+
+impl From<DecryptDeviceFileError> for LoadDeviceError {
+    fn from(value: DecryptDeviceFileError) -> Self {
+        match value {
+            DecryptDeviceFileError::Decrypt(_) => LoadDeviceError::DecryptionFailed,
+            DecryptDeviceFileError::Load(_) => LoadDeviceError::InvalidData,
+        }
+    }
+}
+
+impl From<DecryptDeviceFileError> for ChangeAuthentificationError {
+    fn from(value: DecryptDeviceFileError) -> Self {
+        match value {
+            DecryptDeviceFileError::Decrypt(_) => ChangeAuthentificationError::DecryptionFailed,
+            DecryptDeviceFileError::Load(_) => ChangeAuthentificationError::InvalidData,
+        }
+    }
+}
+
+#[cfg_attr(target_arch = "wasm32", expect(dead_code))]
+fn decrypt_device_file(
+    device_file: &DeviceFile,
+    key: &SecretKey,
+) -> Result<LocalDevice, DecryptDeviceFileError> {
+    let mut cleartext = key
+        .decrypt(device_file.ciphertext())
+        .map_err(DecryptDeviceFileError::Decrypt)
+        .map(zeroize::Zeroizing::new)?;
+    let res = LocalDevice::load(&cleartext).map_err(DecryptDeviceFileError::Load);
+    cleartext.zeroize();
```

Comment:
Isn't the point of `Zeroizing` to automatically `zeroize`?

Reply:
The line was removed